<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>智能家居控制面板</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #f5f5f5;
    font-family: &quot;Microsoft YaHei&quot;, &quot;SimHei&quot;, sans-serif;
}

QGroupBox {
    border: 2px solid #3498db;
    border-radius: 8px;
    margin-top: 10px;
    font-weight: bold;
    color: #2c3e50;
    background-color: white;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 8px 0 8px;
    background-color: #3498db;
    color: white;
    border-radius: 4px;
}

QPushButton {
    border: 2px solid #3498db;
    border-radius: 8px;
    background-color: #3498db;
    color: white;
    font-weight: bold;
    font-size: 14px;
    padding: 8px;
}

QPushButton:hover {
    background-color: #2980b9;
    border-color: #2980b9;
}

QPushButton:pressed {
    background-color: #21618c;
    border-color: #21618c;
}

QPushButton:disabled {
    background-color: #bdc3c7;
    border-color: #bdc3c7;
    color: #7f8c8d;
}

QLineEdit {
    border: 2px solid #bdc3c7;
    border-radius: 6px;
    padding: 6px;
    font-size: 12px;
    background-color: white;
}

QLineEdit:focus {
    border-color: #3498db;
}

QLabel {
    color: #2c3e50;
    font-size: 12px;
}

QLabel[label_voice_result] {
    background-color: #ecf0f1;
    border: 1px solid #bdc3c7;
    border-radius: 4px;
    padding: 4px;
    color: #34495e;
}

QLabel[label_weather_result] {
    background-color: transparent;
    border: none;
    padding: 0px;
    color: #2c3e50;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_server">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>800</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>服务器设置</string>
    </property>
    <widget class="QLabel" name="label_server">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>35</y>
       <width>60</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>服务器：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="hostname">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>32</y>
       <width>150</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>mqtt.yyzlab.com.cn</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_port">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>35</y>
       <width>50</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>端口：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="le_port">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>32</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>1883</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_pub_topic">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>35</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>发布主题：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="pub_topic">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>32</y>
       <width>150</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>1752540016071/APP2AIOTSIM</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_sub_topic">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>35</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>订阅主题：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="sub_topic">
     <property name="geometry">
      <rect>
       <x>640</x>
       <y>32</y>
       <width>141</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>1752540016071/AIOTSIM2APP</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pb_connect">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>20</y>
      <width>90</width>
      <height>35</height>
     </rect>
    </property>
    <property name="text">
     <string>连接</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_weather_result">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>70</y>
      <width>400</width>
      <height>380</height>
     </rect>
    </property>
    <property name="text">
     <string>天气信息将在此显示</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_control">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>800</width>
      <height>301</height>
     </rect>
    </property>
    <property name="title">
     <string>设备控制</string>
    </property>
    <widget class="QLabel" name="label_led_title">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>100</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>灯光控制：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_led">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>80</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>开灯</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_led1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>80</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>LED1</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_led_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>60</y>
       <width>80</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>开灯</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_led2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>150</y>
       <width>80</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>LED2</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_led_3">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>60</y>
       <width>80</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>开灯</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_led3">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>150</y>
       <width>80</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>LED3</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_fan">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>60</y>
       <width>100</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>风扇开</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_fan">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>150</y>
       <width>100</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>风扇</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_beeper">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>60</y>
       <width>100</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>蜂鸣器开</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_beeper">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>150</y>
       <width>100</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>蜂鸣器</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_doorlock">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>60</y>
       <width>100</width>
       <height>80</height>
      </rect>
     </property>
     <property name="text">
      <string>门锁开</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_doorlock">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>150</y>
       <width>100</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>门锁</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_voice_title">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>100</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>语音控制：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pb_voice">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>220</y>
       <width>120</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>按住说话</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_voice_result">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>230</y>
       <width>300</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
      </font>
     </property>
     <property name="text">
      <string>语音识别结果将在此显示</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_sensor">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>440</y>
      <width>1240</width>
      <height>120</height>
     </rect>
    </property>
    <property name="title">
     <string>环境传感器</string>
    </property>
    <widget class="QLabel" name="label_tem">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>280</width>
       <height>60</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>温度：-- ℃</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_hum">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>40</y>
       <width>280</width>
       <height>60</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>湿度：-- %</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_light">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>40</y>
       <width>280</width>
       <height>60</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>光照：-- lux</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_people">
     <property name="geometry">
      <rect>
       <x>930</x>
       <y>40</y>
       <width>280</width>
       <height>60</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>人体红外：无人</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pb_doorlock</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_pb_doorlock_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>340</y>
    </hint>
    <hint type="destinationlabel">
     <x>800</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pb_voice</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_pb_voice_pressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pb_voice</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_pb_voice_released()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
